resources/file.go

```diff
@@ -110,6 +130,32 @@ func (obj *FileRes) Validate() error {
 	return obj.BaseRes.Validate()
 }
 
+func (obj *FileRes) mode() (os.FileMode, error) {
+	m, err := strconv.ParseInt(obj.Mode, 8, 32)
+	if err != nil {
+		return os.FileMode(0), fmt.Errorf("Mode should be an octal number (%v): %v", obj.Mode, err.Error())
+	}
+	return os.FileMode(m), nil
+}
+
+func (obj *FileRes) uid(defaultVal int) (int, error) {
+	if obj.Owner == "" {
+		return defaultVal, nil
+	}
+
+	u, err := user.Lookup(obj.Owner)
```

Comment:
Assuming I'm understand the logic correctly (a comment would help!) we assume the string is a username, and if so return the uid? If not fallback to assuming it's a uid??

If so wouldn't we want to do the direct uid lookup first, as a number could be a username, a username couldn't be an id?